lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -98,3 +108,185 @@ def _validate_and_link(program_uuid_string, linkage_text):
         ]
         errors = [message for message in link_errors.values()]
         return successes, errors
+
+
+class ProgramEnrollmentsInspectorView(View):
+    """
+    The view to search and display the program enrollments
+    information of a learner.
+    """
+    exclude_from_schema = True
+    CONSOLE_TEMPLATE_PATH = 'support/program_enrollments_inspector.html'
+
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        # We need to get some SSO organizations
+        return render_to_response(
+            self.CONSOLE_TEMPLATE_PATH,
+            {
+                'successes': [],
+                'errors': [],
+                'learner_program_enrollments': '',
+                'org_keys': self._get_org_keys_with_IdP_provider(),
+            }
+        )
+
+    @method_decorator(require_support_permission)
+    def post(self, request):
+        """
+        Find the learner and the corresponding ProgramEnrollment info
+        based on the posted organization short name and learner's external user key
+        """
+        errors = []
+        edx_username_or_email = request.POST.get('edx_user', '').strip()
+        org_key = request.POST.get('IdPSelect', '').strip()
+        external_user_key = request.POST.get('external_user_key', '').strip()
+        learner_program_enrollments = {}
+        if edx_username_or_email:
+            learner_program_enrollments, error = self._get_account_info(edx_username_or_email)
+        elif org_key and external_user_key:
+            learner_program_enrollments, error = self._get_account_info_by_external_key(
+                org_key,
+                external_user_key
+            )
+            if error:
+                errors.append(error)
+        else:
+            errors.append(
+                'You must provide either the edX username or email, or the '
+                'Learner Account Provider and External Key pair to do search!'
+            )
+        return render_to_response(
+            TEMPLATE_PATH,
+            {
+                'successes': '',
+                'errors': errors,
+                'learner_program_enrollments': learner_program_enrollments,
+                'org_keys': self._get_org_keys_with_IdP_provider()
+            }
+        )
+
+    def _get_org_keys_with_IdP_provider(self):
+        """
+        From our Third_party_auth models, return a list
+        of organizations whose SAMLProviders are active and configured
+        """
+        orgs_with_saml = []
+        saml_providers = SAMLProviderConfig.objects.current_set().filter(
+            enabled=True,
+            organization__isnull=False
+        ).select_related('organization')
+
+        for saml_provider in saml_providers:
+            orgs_with_saml.append(saml_provider.organization.short_name)
```

Comment:
Do you feel strongly against using a list comprehension for this sort of use case?